### ISSUE TITLE
global: add custom admin views

### ISSUE THREAD

**Issue by krzysztof** [inveniosoftware/invenio-admin#8]
As of yet the entry point discovery is only possible for [flask_admin.contrib.sqla.ModelView](https://flask-admin.readthedocs.org/en/latest/api/mod_contrib_sqla/#flask_admin.contrib.sqla.ModelView). It should be also possible to define and add [flask_admin.base.BaseView](https://flask-admin.readthedocs.org/en/latest/api/mod_base/#flask_admin.base.BaseView)

The entry-point interface doesn't need to change, since the loaded object is a dictionary containing keys 'model' and 'modelview' plus some extras ('category' etc.). To register a BaseView, the InvenioAdmin's init code should check for existence of key 'model' and call the corresponding constructor.

**Comment by jirikuncar** [comment]
Closed via 48be70cc

**Comment by jirikuncar** [comment]
ping @krzysztof (https://github.com/inveniosoftware/invenio-admin/issues/8#issuecomment-239197990)

**Comment by lnielsen** [comment]
Technically it's possible to register a custom view since we have access to the Admin extension, however following is not possible:


```python
# admin.py
from flask_admin import BaseView, expose

class AnalyticsView(BaseView):
    @expose('/')
    def index(self):
        return self.render('analytics_index.html')

analytics = {
    'view_class': AnalyticsView,
    'endpoint': 'analytics',
}

# setup.py
    ...
    entry_points={
        'invenio_admin.views': [
            'analytics = admin:analytics',
        ],
    },
    ...
```

This is due to:

```python
    def register_view(self, view_class, model_class, session=None, **kwargs):
        view_class = self.view_class_factory(view_class)
        self.admin.add_view(
            view_class(model_class, session or db.session, **kwargs))

    def load_entry_point_group(self, entry_point_group):
        for ep in pkg_resources.iter_entry_points(group=entry_point_group):
            admin_ep = dict(ep.load())
            # ...
            self.register_view(
                admin_ep.pop('modelview'),
                admin_ep.pop('model'),
                **admin_ep)
```

I think this is still relevant, but not a blocker for release. Also, changing the dict structure at this point I think requires too many changers in other modules, and a new/old dict structure can easily be supported side-by-side.

**Comment by lnielsen** [comment]
Fixed in b33149707a80433d2b7a9f6b467418f0837678ca
